packages/transactions/src/builders.ts

```diff
@@ -1297,10 +1299,13 @@ export async function callReadOnlyFunction(
 
   const options = Object.assign(defaultOptions, readOnlyFunctionOptions);
 
-  const { contractName, contractAddress, functionName, functionArgs, network, senderAddress } =
+  const { contractName, contractAddress, functionName, functionArgs, network, senderAddress, tip } =
     options;
 
-  const url = network.getReadOnlyFunctionCallApiUrl(contractAddress, contractName, functionName);
+  let url = network.getReadOnlyFunctionCallApiUrl(contractAddress, contractName, functionName);
+  if (tip != null) {
```

Comment:
I know this will check both `null` and `undefined`, but imo it'd be clearer to be explicit and check both. Behaviour of `!=` vs `!==` is a bit of a js quirk that the next developer may not be familiar with. 

Alternatively, why not just flip and condition and check `typeof tip === 'number'`?